### DISCUSSION TITLE
Bin für wemos d1 mini

### DISCUSSION THREAD

**Discussion by interceptor70** [NilsRo/HotWaterRecirculatingPump#2] [Q&A] [answered]
Hallo 

Ich stehe voll auf dem schlauch
 ich bekomme [HotWaterRecirculatingPump](https://github.com/NilsRo/HotWaterRecirculatingPump)

Nicht in meinen wemos d1 mini und wenn ich es doch hoch geladen bekomme 
Startet er nicht mehr ☹ ich Weiß nicht was da los ist
Würde mir bitte jemand eine bin hochladen die ich mit dem EZ-Flasher hochladen kann
Vielen Dank im voraus.

**Comment by NilsRo** [comment]
Moin. 

Das Projekt ist für den ESP32 also z.B. den hier: https://www.fambach.net/d1-mini-esp8266-modul-2-2/

Für den ESP8266 hab ich das nicht ausprobiert, auch wenn es grundsätzlich eigentlich funktionieren müsste. Welchen hast du denn?

Viele Grüße,
Nils

**Comment by interceptor70** [comment]
Ich hab die Berrybase D1-MINI 8266 (baugleich mit den wemos d1 mini)

LG,
Gerrit


------ Originalnachricht ------
Von "NilsRo" ***@***.***>
An "NilsRo/HotWaterRecirculatingPump" 
***@***.***>
Cc "interceptor70" ***@***.***>; "Author" 
***@***.***>
Datum 20.04.2023 22:55:35
Betreff Re: [NilsRo/HotWaterRecirculatingPump] Bin für wemos d1 mini 
(Discussion #2)

**Reply by NilsRo** [reply]
Dann musst das umbauen und selber kompilieren. Das Image hier ist nur auf dem ESP32 lauffähig. Die Libraries müssten eigentlich alle kompatibel sein...

**Reply by interceptor70** [reply]
ja sind sie :-)

versuche es jetzt in FHEM einzubinden :)

**Comment by NilsRo** [comment]
Wenn du magst mach gerne ein Pull Request, kann es leider aufgrund fehlender Hardware nicht selber testen.

**Reply by interceptor70** [reply]
Also portieren hat funktioniert ist aber leider viel zu wenig Speicher beim D1-mini 8622

Deshalb habe ich doch den Großen Bruder genommen. 

Ist jetzt genau so aufgebaut wie bei dir beschrieben :-)

**Reply by NilsRo** [reply]
Wenn noch was nicht funktioniert oder was brauchst sag gerne Bescheid. Bin auch gerade nochmal dran gewesen, schau da gerne ob die aktuelle Version erwischt hast. 

Mit den Bibliotheken und allem ist es für die einfache Logik nicht gerade sparsam ausgelegt. 😅 Aber privat macht das ja nicht viel Unterschied.

**Comment by interceptor70** [comment]
Ein Schaltplan im Eagle Format währe super, mit KiCAD bin ich nicht so auf der Höhe …. :)

**Reply by NilsRo** [reply]
Mit Eagle hab ich mal angefangen mir Schaltpläne selber beizubringen, nachdem das nicht mehr weiterentwickelt wird bin ich direkt auf KiCad. 

Unter Docs hab ich den Schaltplan auch als PDF abgelegt.

**Reply by interceptor70** [reply]
dann passt aber leider das gehäuse nicht mehr

**Reply by NilsRo** [reply]
Was meinst denn damit? Lochrasterplatine hab ich Adhoc gemacht.

**Reply by interceptor70** [reply]
Ach so …. Dann baue ich es auch auf nen lochraster auf dache wegen der Größe von der Platine :)

**Reply by NilsRo** [reply]
Das passt ganz gut. Hab's mit Kicad leider nicht hinbekommen das als Lochrasterplatine zu erstellen und dann einfach gemacht.

**Comment by interceptor70** [comment]
Kann ich die Pumpe via command starten … oder den Modus wechseln ?

Oder gibt es schon eine fhem Template ?

**Reply by NilsRo** [reply]
Gibt 3 MQTT Subscriptions anzugeben:

thermal desinfection topic: geführte Schaltung, läuft so lange bis eine andere Nachricht kommt
external pump start topic: manueller trigger einmal das Wasser umzuwälzen (hab ich per Bewegungsmelder wenn ich aufstehe)
heater status topic: schaltet die Automatik an/aus wenn z.B: die Heizung abschaltet oder man in Urlaub ist. Wie man es hinterlegt...

FHEM nutze ich nicht, daher auch kein Template. Aber gerne eines erstellen. :-)

**Reply by interceptor70** [reply]
Ok ich versuche mein glück :-)

**Reply by NilsRo** [reply]
Hab ein neues Release gemacht, die Desinfektion hat nicht richtig funktioniert. Eine bin hängt an, kannst auch direkt im Webinterface hochladen. Du solltest die Version 0.91 nutzen, jetzt sollte erstmal alles passen.

**Reply by NilsRo** [reply]
Zum Testen vor Ort kannst auch den Taster gedrückt halten...

**Comment by interceptor70** [comment]
Was ist das eigentlich für ein fehler ?

Could not find the any filesystem section in the partitions table

**Reply by NilsRo** [reply]
Noch nie gesehen. Welches Board verwendest du genau? Wann kommt der?

Hab das Core Dump Feature eingebaut welches einen Dump ins Flash wegschreibt. Ggf. gibt es da Probleme.

**Reply by interceptor70** [reply]
ok mein fehler ..... datei vergessen :-(

**Comment by interceptor70** [comment]
FHEM Schickt sheinbar alles ins nirvana .....

$DEVICETOPIC:ww/ht/pump 1 Kommt nicht an bzw reagiert nicht

**Reply by NilsRo** [reply]
Was sagt denn ein Monitoring Programm wie MQTT Explorer? Kommen die sendenden Topics in FHEM an?

In den Einstellungen ist das Topic gesetzt? Vertraue da der Bibliothek nicht ganz.

**Comment by interceptor70** [comment]
Also so würde das bei FHEM aussehen :

```
defmod Heizung MQTT2_DEVICE esp32_XXXXXXXXXXXX
attr Heizung autocreate 1
attr Heizung devStateIcon {my $vV = ReadingsVal("Heizung", "dhw_Tflow_measured", "30");; \
my $colVor = substr(Color::pahColor(20,40,60,$vV,0),0,6);; \
my $iconVor = 'sani_supply_temp@'.$colVor;; \
my $vR = ReadingsVal("Heizung", "dhw_Treturn", "30");; \
my $colRueck = substr(Color::pahColor(20,40,60,$vR,0),0,6);; \
my $iconRueck = 'sani_return_temp@'.$colRueck;; \
my $vW = ReadingsVal("Heizung", "Tint", "30");; \
my $colWarm = substr(Color::pahColor(20,40,60,$vW,0),0,6);; \
my $iconWarm = 'sani_water_hot@'.$colWarm;; \
my $vA = ReadingsVal("Heizung", "dhw_pump_circulation", "30");; \
my $colAussen = substr(Color::pahColor(-10,10,30,$vA,0),0,6);; \
my $iconAussen = 'sani_pump@'.$colAussen;; ;; \
\
"<div style=\"text-align:right\" > Vorlauf: " . FW_makeImage("$iconVor",'file_unknown@grey') . " $vV °C<br>Ruecklauf: " . FW_makeImage("$iconRueck",'file_unknown@grey') . " $vR °C<br>Kessel: " . FW_makeImage("$iconWarm",'file_unknown@grey') . " $vW °C<br>Circulation: " . FW_makeImage("$iconAussen",'file_unknown@grey') . " " . int($vA*10)/10 . " </div>"}
attr Heizung devicetopic ww/ht
attr Heizung icon Heizung_Hauptschalter.on
attr Heizung readingList esp32_XXXXXXXXXXXX:ww/ht/info:.* info\
esp32_XXXXXXXXXXXX:ww/ht/dhw_pump_circulation:.* dhw_pump_circulation\
esp32_XXXXXXXXXXXX:ww/ht/dhw_Tflow_measured:.* dhw_Tflow_measured\
esp32_XXXXXXXXXXXX:ww/ht/dhw_Treturn:.* dhw_Treturn\
esp32_XXXXXXXXXXXX:ww/ht/Tint:.* Tint
attr Heizung room Heizung,MQTT2_DEVICE
```

![FHEM-Heizung](https://user-images.githubusercontent.com/35465613/234873881-aa30a248-9d42-40c8-867a-c07bc50f5ccd.png)

Ich muss jetzt nur noch rausbekommen welchen befehl er ESP erwartet für den modus


und ich habe noch was gemacht :-)
![HWR](https://user-images.githubusercontent.com/35465613/234897841-30319756-f602-461f-b664-a02d5482591f.png)

**Reply by NilsRo** [reply]
Professionell! 😃 Eagle oder Kicad?

Wenn die Nachrichten ankommen kann es nur sein, dass FHEM die nicht sendet oder der Datentypen nicht stimmt. Wenn das im MQTT Trace siehst versuch mal einen String. Wenn FHEM eine echte Zahl sendet würde das nicht klappen.

**Reply by interceptor70** [reply]
Online mit EasyEDA :-)

**Reply by interceptor70** [reply]
```
off:noArg    cmnd/esp32_XXXXXXXXXXXX/pump 0
  on:noArg     cmnd/esp32_XXXXXXXXXXXX/pump 1
  toggle:noArg cmnd/esp32_XXXXXXXXXXXX/pump 2
```

**Reply by NilsRo** [reply]
> Online mit EasyEDA :-)

Kurz noch ein paar Hinweise: 

einen DS18B20 hab ich für die interne Temperaturmessung vorgesehen der kann direkt aufs Board. War mir wegen der fehlenden Belüftgung nicht sicher... 

Wifi-Button brauchst eigentlich nie, ist nur zum Zurücksetzen der Wifi Settings wenn der WLan findet und man manuell zurücksetzen möchte oder man sein Passwort vergessen hat. Für den Fall, dass er das Wlan nicht findet geht er automatisch in den Access Point Modus zur Konfiguration. 

Ohne Button auf dem Board kannst das Gehäuse nicht ohne Anpassung nutzen.

**Reply by interceptor70** [reply]
Ok das ist gut zu wissen, danke für die Info :-)

aber nichts desto trotz bekomme ich die Pumpe nicht gestartet :-(

![MQTT](https://user-images.githubusercontent.com/35465613/235306902-1330e72c-6db6-4342-8647-b26fbecbbfd5.jpg)

**Reply by interceptor70** [reply]
ok doch gefunden hab den Wald vor lauter Bäumen nicht gesehen.

Jetzt währe toll wenn man den Status der Pumpe auch noch über mqtt bekommen würde
und wenn ich sie stoppen könnte.

**Reply by NilsRo** [reply]
Das ist das Topic ob die Pumpe läuft: ww/ht/dhw_pump_circulation

Im Automatikbetrieb geht sie von selber aus wenn alles umgewälzt ist. Du kannst sie aber auch manuell an-/ausschalten (siehe die 3 subscriptions in der README. Macht aber meistens keinen Sinn, ich würde es lassen.

**Reply by interceptor70** [reply]
ja ist richtig.

Ich meine  
 die Status Meldung z.b.   "heater off" oder "desinfection" und "automatic" bekomme ich irgendwie nicht zurück das meine ich damit

**Reply by NilsRo** [reply]
Verstehe, hab ein Topic "status" ergänzt. Macht Sinn das zu haben, die README etc. passe ich die Tage an, kannst es aber schon aus dem Main-Branch ziehen.

**Reply by interceptor70** [reply]
Komisch ist nur das ab und an > Update error: Partition Could Not be Found < auftaucht wo kommt das her ?

**Reply by NilsRo** [reply]
Wenn's nur beim OTA Update kommt ist es irgendwas dort. Der nutzt eine Partition dafür und da geht wohl was schief. Ich hab's noch nicht gehabt, ignorier es erstmal. Oder kommt es auch im normalen Betrieb?

**Comment by interceptor70** [comment]
Could not find the any filesystem section in the partitions table c:\HotWaterRecirculatingPump-1\partitions.csv

wenn ich es kompilieren will als ota file kommt das auch.

**Comment by interceptor70** [comment]
Schon richtig toll. gefällt mir super.

Aber warum bekomme ich die pumpe manuell nicht gestartet bzw. gestoppt ?

Alles andere funktioniert jetzt super

**Comment by NilsRo** [comment]
Schick mal was du eingestellt hast für die Topics in der Konfiguration. Oben hattest du ja einen Screenshot vom MQTT Explorer gepostet, /ht/ww/Pumpe war mit einer 1 versorgt, oder? Du kannst das auch im MQTT Explorer testen und dort einen Wert senden.

**Reply by interceptor70** [reply]
habe es etwas geändert auf on 👍 

modus :
desinfection startet -> sofort
urlaub                      -> sofort
auto                         -> sofort
pumpe                      -> keine reaktion



![neu](https://user-images.githubusercontent.com/35465613/235509946-a61f45e8-bf21-4db0-898d-eff0dbe70591.jpg)

**Reply by interceptor70** [reply]
MQTT Explorer getesten und dort einen Wert gesendet -> leider keine reaktion

**Reply by NilsRo** [reply]
Nur zur Sicherheit, im Explorer stand Pumpe, das topic ist aber dhw_pumpe...hast wahrscheinlich angepasst aber schau mal. Die empfangenen Werte werden auch über Serial ausgegeben. 

Und ich hab noch die Frage, warum du die Pumpe ausschalten möchtest. Das Projekt soll genau sowas überflüssig machen...

**Reply by interceptor70** [reply]
Zu Wartung dachte ich wäre das sinnvoll, ein art Wartungs Modus zu haben der alles deaktiviert.

![term](https://user-images.githubusercontent.com/35465613/235531749-ca0a56cc-426e-4d58-a5ce-8335fcde0c06.jpg)

also das kommt im term. an

**Reply by NilsRo** [reply]
Lass mal die Heizung an. Dann sollte es gehen, macht aber mehr Sinn die Pumpe immer zu starten denke ich. Muss ich mal ändern. Im Warmwasserspeicher kann ja genug Restwärme sein. 

Ich nenne die Beschreibungen vielleicht besser noch um. 'Heather off' bedeutet nur, dass die Automatik nicht mehr läuft weil die Heizung in der Nachtabsenkung ist. Das ist für Urlaube oder die Nacht praktisch, um Fehlerkennungen auszuschließen. Aktuell wird aber noch der manuelle Trigger geblockt...

Wartungsmodus gibt es aktuell nur über den Button, wenn man den länger drückt.

**Reply by interceptor70** [reply]
ja das ist auch super, wenn man die Pumpe testen möchte ob sie läuft.

Sind aber Wartungen an der Heizung ist das glaube ich nicht so toll, wenn die pumpe
Einfach anspringt oder ?

**Reply by NilsRo** [reply]
Stromlos machen. Alle 24h spült er auch um Legionellen, etc. zu vermeiden ist egal was ist. Das würde ich nicht ausschaltbar machen wollen wenn jemand vergisst die Automatik wieder zu aktivieren. Den Trigger fixe ich die Tage...

**Reply by interceptor70** [reply]
Jetzt habe ich noch ein Problem, weil ich die Pumpe nicht stoppen kann.
 Er hängt scheinbar bei der Desinfektion fest warum auch immer 

Und wenn man den Wartungs Modus auf zeit begrenzt z.b. inerhalb 60 minuten nochmal taste drücken um die Wartung zu verlängern, währe doch auch eine gute Lösung oder ?

**Reply by NilsRo** [reply, marked as answer]
Hab im Branch "Development" die Änderung gemacht und noch Anpassungen am Status. Schau es dir gerne an.
Die Desinfektion wartet wie ein normaler Lauf, dass das Wasser einmal komplett umgewälzt wurde läuft nur unendlich weiter bis diese über eine andere Nachricht gelöst wird. 

Heißt, wenn du die kurz an und wieder aus machst wird so lange gepumpt bis die notwendige Rücklauftemperatur erreicht ist.

**Comment by NilsRo** [comment]
Hab eine neue Version veröffentlich mit Json als Status. Dann kann ich das ohne Probleme erweitern. Ich hoffe es läuft nun alles bei dir.
